examples/instantsearch/src/instantsearch.ts

```diff
@@ -0,0 +1,187 @@
+
+          <div
+            style="
+              display: grid;
+              grid-auto-flow: column;
+              justify-content: start;
+              align-items: center;
+              gap: 8px;
+            "
+          >
+            {{#rating}}
+              <div
+                style="
+                  display: grid;
+                  grid-auto-flow: column;
+                  justify-content: start;
+                  align-items: center;
+                  gap: 4px;
+                "
+              >
+                <svg
+                  width="16"
+                  height="16"
+                  viewBox="0 0 24 24"
+                  fill="#e2a400"
+                  stroke="#e2a400"
+                  stroke-width="1"
+                  stroke-linecap="round"
+                  stroke-linejoin="round"
+                >
+                  <polygon points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2" />
+                </svg>
+                {{rating}}
+              </div>
+            {{/rating}}
+
+            <div
+              style="
+                display: grid;
+                grid-auto-flow: column;
+                justify-content: start;
+                align-items: center;
+                gap: 4px;
+              "
+            >
+              <svg
+                width="16"
+                height="16"
+                viewBox="0 0 24 24"
+                fill="none"
+                stroke="currentColor"
+                stroke-width="2"
+                stroke-linecap="round"
+                stroke-linejoin="round"
+                style="
+                  position: relative;
+                  top: 1px;
+                "
+              >
+                <path d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2z"></path>
+              </svg>
+              <span>{{comments}}</span>
+            </div>
+          </div>
+        </article>
+      `,
+    },
+  }),
+  pagination({
+    container: '#pagination',
+    padding: 2,
+    showFirst: false,
+    showLast: false,
+  }),
+]);
+
+export function setInstantSearchUiState(indexUiState) {
+  search.setUiState((uiState) => ({
+    ...uiState,
+    [instantSearchIndexName]: {
+      ...uiState[instantSearchIndexName],
+      page: 1,
+      ...indexUiState,
+    },
+  }));
+}
+
+export const debouncedSetInstantSearchUiState = debounce(
+  setInstantSearchUiState,
+  500
+);
+
+export function getInstantSearchActiveCategory() {
+  const indexUiState = search.renderState.instant_search;
+  const hierarchicalMenuUiState = indexUiState && indexUiState.hierarchicalMenu;
+  const categories =
+    (hierarchicalMenuUiState &&
+      hierarchicalMenuUiState[hierarchicalAttribute] &&
+      hierarchicalMenuUiState[hierarchicalAttribute].items) ||
+    [];
+  const refinedCategory = categories.find((category) => category.isRefined);
+
+  return refinedCategory && refinedCategory.value;
```

Comment:
this would be easier to compute from the uistate, something approximately like this

```js
search.mainIndex.getWidgetUiState().hierarchicalMenu?.[hierarchicalAttribute]
```

Reply:
I'd like to avoid going deep into nested internals with `search.mainIndex.getWidgetUiState()` as opposed to `search.renderState`.

Reply:
That makes sense, and makes me think we should expose getUiState at the root level, since the use case makes more sense with ui state than with render state.